test/test_losses.py

```diff
@@ -0,0 +1,226 @@
+    box2 = torch.tensor([0, 0, 1, 1], dtype=dtype, device=device)
+    box3 = torch.tensor([0, 1, 1, 2], dtype=dtype, device=device)
+    box4 = torch.tensor([1, 1, 2, 2], dtype=dtype, device=device)
+
+    box1s = torch.stack([box2, box2], dim=0)
+    box2s = torch.stack([box3, box4], dim=0)
+
+    return box1, box2, box3, box4, box1s, box2s
+
+
+def assert_iou_loss(iou_fn, box1, box2, expected_loss, dtype, device, reduction="none"):
+    computed_loss = iou_fn(box1, box2, reduction=reduction)
+    expected_loss = torch.tensor(expected_loss, device=device)
+    torch.testing.assert_close(computed_loss, expected_loss)
+
+
+def assert_empty_loss(iou_fn, dtype, device):
+    box1 = torch.randn([0, 4], dtype=dtype, device=device).requires_grad_()
+    box2 = torch.randn([0, 4], dtype=dtype, device=device).requires_grad_()
+    loss = iou_fn(box1, box2, reduction="mean")
+    loss.backward()
+    torch.testing.assert_close(loss, torch.tensor(0.0, device=device))
+    assert box1.grad is not None, "box1.grad should not be None after backward is called"
+    assert box2.grad is not None, "box2.grad should not be None after backward is called"
+    loss = iou_fn(box1, box2, reduction="none")
+    assert loss.numel() == 0, f"{str(iou_fn)} for two empty box should be empty"
+
+
+class TestGeneralizedBoxIouLoss:
+    # We refer to original test: https://github.com/facebookresearch/fvcore/blob/main/tests/test_giou_loss.py
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.half])
+    def test_giou_loss(self, dtype, device):
+
+        box1, box2, box3, box4, box1s, box2s = get_boxes(dtype, device)
+
+        # Identical boxes should have loss of 0
+        assert_iou_loss(ops.generalized_box_iou_loss, box1, box1, 0.0, dtype=dtype, device=device)
+
+        # quarter size box inside other box = IoU of 0.25
+        assert_iou_loss(ops.generalized_box_iou_loss, box1, box2, 0.75, dtype=dtype, device=device)
+
+        # Two side by side boxes, area=union
+        # IoU=0 and GIoU=0 (loss 1.0)
+        assert_iou_loss(ops.generalized_box_iou_loss, box2, box3, 1.0, dtype=dtype, device=device)
+
+        # Two diagonally adjacent boxes, area=2*union
+        # IoU=0 and GIoU=-0.5 (loss 1.5)
+        assert_iou_loss(ops.generalized_box_iou_loss, box2, box4, 1.5, dtype=dtype, device=device)
+
+        # Test batched loss and reductions
+        assert_iou_loss(ops.generalized_box_iou_loss, box1s, box2s, 2.5, dtype=dtype, device=device, reduction="sum")
+        assert_iou_loss(ops.generalized_box_iou_loss, box1s, box2s, 1.25, dtype=dtype, device=device, reduction="mean")
+
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.half])
+    def test_empty_inputs(self, dtype, device):
+        assert_empty_loss(ops.generalized_box_iou_loss, dtype, device)
+
+
+class TestCIOULoss:
```

Comment:
Minor detail: here the class name follows a different convetion `IOU` vs `Iou`. Better have everything following the same, so for example `IoU`.